src/main/java/wolox/training/exceptions/BookIdMismatchException.java

```diff
@@ -0,0 +1,5 @@
+package wolox.training.exceptions;
+
+public class BookIdMismatchException extends RuntimeException {
+
```

Comment:
I had to remove the constructor they recommended because it wouldn't work with the `::new` thing. Which one is correct? Also, I used `RuntimeException`s but what's the standard really? It used to be "checked exceptions are bad" when I used Java.

Reply:
RuntimeExceptions are `unchecked` actually. Exception is the class that is `checked` :)

Reply:
The :: operator is useful when calling parameter-less methods, such as the default constructor.
If you need to call a constructor that takes arguments, you have two options:
Either you use the usual way of instantiating a class (i.e. new MyClass(args)), or you can use what [this post](https://www.baeldung.com/java-8-double-colon-operator)  describes
Basically you use the :: operator to generate a "Function" that afterwards you use to pass the actual parameters.
Given in these cases you might want to pass the arguments to the exception in order to set the exception message for example, I would restore the constructor that you had and call it in any of the possible ways.

Reply:
That's really neat, thanks

Reply:
And yes, what I meant was that the standard was to only use RuntimeException, no checked exceptions